components/admin_console/license_settings/license_settings.jsx

```diff
@@ -61,11 +65,12 @@ export default class LicenseSettings extends React.PureComponent {
         this.fileInputRef = React.createRef();
     }
 
-    componentDidMount() {
+    async componentDidMount() {
         if (!this.props.enterpriseReady) {
             this.reloadPercentage();
         }
         this.props.actions.getLicenseConfig();
+        this.props.actions.getPrevTrialLicense();
```

Comment:
We can probably send these requests in a promise all with an `await`?

Reply:
We can do that but that won't be the redux way (action -> reducer -> state).  That's why we didn't consume the function response.

Reply:
You don't need to consume it to use Promise.all. You can use the following example to do what ben is saying
This example was pulled from `group_details.jsx`
```js
        await Promise.all([
            this.props.actions.getGroupSyncables(this.props.groupID, Groups.SYNCABLE_TYPE_CHANNEL),
            this.props.actions.getGroupSyncables(this.props.groupID, Groups.SYNCABLE_TYPE_TEAM),
        ]);
```

Reply:
It'll run all those actions in parallel and it doesn't need to wait one by one to finish and go on to the next

Reply:
We just do these actions in parallel and don't need to consume it still gets updated in state.

Reply:
As discussed with @hahmadia, we don't need t use `promise.all` as `getLicenseConfig` and `getPrevTrialLicense ` are already non-blocking.